Add 2i-approved release notes content to changelog (#2473)
alphagov/govuk-frontend: CHANGELOG.md

## Unreleased

### Breaking changes

> This release contains a lot of breaking changes, but we expect that many of them will only affect a small number of users. However, to make sure your service still works after you upgrade, you should read the release notes and make any required changes.

Author: ```suggestion
This release contains a lot of breaking changes, but we expect many of them will only affect a small number of users. However, to make sure your service still works after you upgrade, you should read the release notes and make any required changes.
```

> This release contains a lot of breaking changes, but we expect that many of them will only affect a small number of users. However, to make sure your service still works after you upgrade, you should still read the release notes and make any required changes.

Author: ```suggestion
This release contains a lot of breaking changes, but we expect that many of them will only affect a small number of users. However, to make sure your service still works after you upgrade, you should read the release notes and make any required changes.
```

> This release contains a lot of breaking changes, but we expect that many of them will only affect a small number of users. However, to make sure your service continues to work after you upgrade, you should still read the release notes and make any required changes.

Author: ```suggestion
This release contains a lot of breaking changes, but we expect that many of them will only affect a small number of users. However, to make sure your service still works after you upgrade, you should still read the release notes and make any required changes.
```

#### Check your accordions still display as expected

> We've changed the design of the accordion. [See its new look, and read the revised guidance, particularly about when, and when not, to use accordions](https://design-system.service.gov.uk/components/accordion/).

Author: ```suggestion
We've changed the design of the accordion. You can [see its new look, and read the revised guidance, particularly about when, and when not, to use accordions](https://design-system.service.gov.uk/components/accordion/).
```

> You should check that your accordions still display as expected if you:

Author: ```suggestion
You should check your accordions still display as expected if you:
```

## Unreleased

### Breaking changes
This release contains a lot of breaking changes, but we expect many of them will only affect a small number of users. However, to make sure your service still works after you upgrade, you should read the release notes and make any required changes.

#### Check your accordions still display as expected

We've changed the design of the accordion. You can [see its new look, and read the revised guidance, particularly about when, and when not, to use accordions](https://design-system.service.gov.uk/components/accordion/).

You should check your accordions still display as expected if you:

- have applied any custom styles to your accordions

> - use accordions in non-standard ways (for example, to display accordions, other components, tables or large images)

Author: @hannalaakso Does this work?

```suggestion
- have modified the standard Design System accordion (for example, by adding custom styles)
```

Check your changes against the [header example in the Design System](https://design-system.service.gov.uk/components/header/#header-with-service-name-and-navigation).

> These changes were introduced in [pull request #2427: Improve the screen reader experience for the header menu](https://github.com/alphagov/govuk-frontend/pull/2427) and [pull request #2471: Add missing instructions for header to changelog](https://github.com/alphagov/govuk-frontend/pull/2471).

Author: @36degrees Is this the right place to mention #2471? Added it here, following [your comment in the Google Doc re making sure we don't lose changes](https://docs.google.com/document/d/1atVBJI2ETBskZay_coZJxEsL1T4BOgoKY9BKszyr4Bo/edit?disco=AAAAS2EIacI).

> These changes were introduced in [pull request #2427: Improve the screen reader experience for the header menu](https://github.com/alphagov/govuk-frontend/pull/2427) and [pull request #2471: Add missing instructions for header to changelog](https://github.com/alphagov/govuk-frontend/pull/2471).

Author: ```suggestion
This change was introduced in [pull request #2427: Improve the screen reader experience for the header menu](https://github.com/alphagov/govuk-frontend/pull/2427) and [pull request #2471: Add missing instructions for header to changelog](https://github.com/alphagov/govuk-frontend/pull/2471).
```

> This change was introduced in [pull request #2427: Improve the screen reader experience for the header menu](https://github.com/alphagov/govuk-frontend/pull/2427).

Author: Do we need to mention [#2471: Add missing instructions for header to changelog](https://github.com/alphagov/govuk-frontend/pull/2471)?

Author: @36degrees @vanitabarrett

Author: No need to mention this, will delete this comment.

```html
<div class="govuk-skip-link" data-module="govuk-skip-link">
...
</div>
```

If you're [importing JavaScript for individual components](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#select-and-initialise-an-individual-component), import the skip link JavaScript.

Once you've made the changes, check that the skip link JavaScript works. To make sure, click the skip link and check that the linked element (usually the `<main>` element) in the browser has a `tabindex` attribute.

This change was introduced in [pull request #2450: Set focus to skip link target to improve screen reader announcements](https://github.com/alphagov/govuk-frontend/pull/2450).

#### Update the HTML for error messages

We’ve updated the HTML for the error message component to use a paragraph element instead of a span element, as this is more semantically correct.

If you’re not using Nunjucks macros, swap the `<span class="govuk-error-message">` for a `<p class="govuk-error-message">`.

This change was introduced in [pull request #2452: Update error messages to use paragraph tags instead of spans](https://github.com/alphagov/govuk-frontend/pull/2452).

> #### Import 'base' before importing Sass files from `core` or `overrides` layers

Reviewer: @36degrees @EoinShaughnessy Do you think it's worth putting this one and the one underneath together with "#### Make sure individually imported JavaScript modules work as expected"?

Author: Yeah, I think that sounds good. Have moved. So now, after 'Make sure individually imported JavaScript modules work as expected' (line 121), we have these sections:

- Import 'base' before importing Sass files from `core` or `overrides` layers (line 144)
- Import 'template' Sass file from the objects layer instead of the core layer (line 157)

Let me know if that's ok, or if there's a reason to change.

On radios and checkboxes, the JavaScript now looks within the whole page for conditionally-revealed content. Before, it only looked within the same set of radios or checkboxes.

> If you see unexpected behaviour, [make sure the revealed content's `id` is unique within the page the content is on](https://www.w3.org/WAI/WCAG21/Techniques/html/H93). Reusing the same `id` within a page could cause a radio or checkbox to be revealed and hide the wrong element, and also means your HTML is invalid.

Author: Thanks Vanita! Might revert in that case?

```suggestion
If you see unexpected behaviour, [make sure the revealed content's `id` is unique within the page the content is on](https://www.w3.org/WAI/WCAG21/Techniques/html/H93). Reusing the same `id` within a page could cause a radio or checkbox to reveal and hide the wrong element, and also means your HTML is invalid.
```

> If you see unexpected behaviour, [make sure the revealed content's `id` is unique within the page the content is on](https://www.w3.org/WAI/WCAG21/Techniques/html/H93). Reusing the same `id` within a page could cause a radio or checkbox to be revealed and hide the wrong element, and also means your HTML is invalid.

Author: @vanitabarrett @36degrees Feedback from 2i reviewer:

> Could we put 'reveal or hide'? I think the current wording makes it sound as if you can reveal and hide a certain element at the same time.
